spec/integration/helper.rb

```diff
@@ -0,0 +1,452 @@
+    if tags[:distribution].is_a? Array
+      filter = tags[:distribution]
+    else
+      filter = [tags[:distribution]]
+    end
+    versions &= filter
+  end
+
+  # Check if the Portus image has to be rebuilt.
+  rebuild = true
+  rebuild = tags[:rebuild] == true if tags.key?(:rebuild)
+
+  [versions, rebuild]
+end
+
+# Execute the given block just once.
+def once(id, &blk)
+  blk.call if ENV["INTEGRATION_ONCE_BLOCK_#{id}"] != "t"
+  ENV["INTEGRATION_ONCE_BLOCK_#{id}"] = "t"
+end
+
+# Returns whether LDAP has to be run on the current `integration` block.
+def ldap?
+  ENV["INTEGRATION_LDAP"] == "t" || ENV["INTEGRATION_LDAP"] == "true"
+end
+
+# Makes sure that there is a container running the given Docker distribution
+# version.
+def ensure_distribution!(version)
+  name = "portus_distribution_#{version.gsub(".", "")}"
+
+  src  = File.expand_path(File.dirname(__FILE__)) + "/" + ("fixtures")
+  img  = "registry:#{version}"
+  opts = {
+    "Volumes"      => { src => { "/etc/docker/registry" => "ro,Z", "/registry_data" => "" } },
+    "ExposedPorts" => { "5000/tcp" => {} },
+    "HostConfig"   => {
+      "Binds"        => ["#{src}:/etc/docker/registry", "#{src}/data:/registry_data"],
+      "Links"        => ["integration_portus"],
+      "PortBindings" => { "5000/tcp" => [{ "HostIP" => "0.0.0.0", "HostPort" => "5000" }] }
+    }
+  }
+
+  start_container!(img, name, opts)
+end
+
+def reset_db!
+  mysql_ping!
+
+  # Migrate & seed
+  docker_exec("integration_portus", "rake db:drop")
+  docker_exec("integration_portus", "rake db:create")
+  docker_exec("integration_portus", "rake db:migrate:reset")
+  docker_exec("integration_portus", "rake db:seed")
+
+  # Introduce the current registry.
+  cmd = "Registry.create(name: \\\"registry\\\", hostname: \\\"#{registry_hostname}\\\", "\
+    "use_ssl: false)"
+  docker_exec("integration_portus", "rails runner \"#{cmd}\"")
+end
+
+# Returns the hostname of the current registry.
+def registry_hostname
+  "#{ip}:5000"
+end
+
+# Start a container from the given image. It accepts a hash with extended
+# options that you might want to pass it on creation time.
+def start_container!(image, name, ext = {})
+  # Remove old containers.
+  container = container_for(name)
+  container.delete(force: true) if container
+
+  opts = { "Image" => image }.merge(ext)
+
+  begin
+    container = Docker::Container.create(opts)
+  rescue Docker::Error::NotFoundError
+    puts "Pulling from #{image}"
+    pull(image)
+    container = Docker::Container.create(opts)
+  end
+
+  container.rename(name)
+  puts "Starting container #{name}"
+  container.start
+
+  # Make sure that the container has actually started. Yes, the previous
+  # `start` method does not say anything about this (and even worse, we don't
+  # have a way in the client to tell why the container could not start).
+  begin
+    container.top
+  rescue Docker::Error::ServerError
+    raise StartError, "Could not start container."
```

Comment:
Maybe we should add the name of the image to make the debugging message even more useful